src/main/scala/ai/privado/languageEngine/ruby/tagger/collection/CollectionTagger.scala

```diff
@@ -0,0 +1,132 @@
+package ai.privado.languageEngine.ruby.tagger.collection
+
+import ai.privado.cache.RuleCache
+import ai.privado.model.{Constants, InternalTag, RuleInfo}
+import ai.privado.tagger.PrivadoParallelCpgPass
+import ai.privado.utility.Utilities.*
+import io.shiftleft.codepropertygraph.generated.Cpg
+import io.shiftleft.codepropertygraph.generated.nodes.{Call, Method}
+import io.shiftleft.semanticcpg.language.*
+import org.slf4j.LoggerFactory
+
+import scala.collection.mutable
+import scala.util.{Failure, Success, Try}
+
+class CollectionTagger(cpg: Cpg, ruleCache: RuleCache) extends PrivadoParallelCpgPass[RuleInfo](cpg) {
+  private val logger       = LoggerFactory.getLogger(this.getClass)
+  private val methodUrlMap = mutable.HashMap[Long, String]()
+  private val classUrlMap  = mutable.HashMap[Long, String]()
+
+  override def generateParts(): Array[RuleInfo] =
+    ruleCache.getRule.collections.filter(_.catLevelTwo == Constants.default).toArray
+
+  override def runOnPart(builder: DiffGraphBuilder, collectionRuleInfo: RuleInfo): Unit = {
+
+    val collectionMethods = cpg
+      .call("get|post|put|patch|delete")
+      .where(_.file.name(".*routes.rb"))
+      .filter(_.argument.isCall.code("to.*").argument.isLiteral.nonEmpty)
```

Comment:
The regex can be `to:.*` as we are using the same below as well. Better to create a local constant regex and use it at both places

Reply:
Handled here: 30ba3e8319958e033e164959efc86e06b4c9ddab